internal/backend/remote-state/http/client_test.go

```diff
@@ -97,77 +93,12 @@ func TestHTTPClient(t *testing.T) {
 	remote.TestClient(t, client)
 }
 
-type testHTTPHandler struct {
```

Comment:
Ah you found it! I like the solution of keeping a mock here and reusing it in the new tests 👌🏻